modules/assim.sequential/R/Remote_helpers.R

```diff
@@ -0,0 +1,411 @@
+  #---------------------------------------------------------------
+  # copying over the luncher
+  remote.copy.to(
+    my_host,
+    system.file("RemoteLauncher", "SDA_launcher.R", package = "PEcAn.assim.sequential"),
+    settings$outdir,
+    delete = FALSE,
+    stderr = FALSE
+  )
+  
+  cmd <- paste0("nohup  Rscript ",
+                settings$outdir,"//SDA_launcher.R ", # remote luncher
+                settings$outdir,"//",basename(settingPath), # path to settings
+                " Obs//", basename(ObsPath), # Path to Obs
+                " > ", 
+                settings$outdir,"/SDA_remote_nohup.out 2>&1 &"
+  )
+  
+
+
+   PEcAn.logger::logger.info("Running this command on your remote: \n")
+   PEcAn.logger::logger.info(cmd)
+  
+   
+   #calling SDA
+   out<-remote.execute.R(paste0("system(\" ",cmd, "\")"),
+                         my_host,
+                         user = my_host$user,
+                         scratchdir = ".")
+   
+   
+   
+   # Let's see what is the PID of the job doing the nohup
+   # I'll use this to track the progress of my SDA job
+   PIDS<-remote.execute.cmd(my_host, cmd = "lsof",
+                      args = c(paste0(settings$outdir,"/SDA_remote_nohup.out")))
+   
+   if (length(PIDS)>1){
+     #some cleaning
+     PID<-PIDS[-1] %>% 
+       map_dbl(function(line){
+         ll <- strsplit(line, " ")[[1]]
+         ll <- ll[nchar(ll)>0]
+         ll[2] %>% 
+           as.numeric()
+       }) %>%
+       unique()
+   }else{
+     PEcAn.logger::logger.severe("Something broke the run before it starts!")
+     #TODO: read the nohup.out if it exists
+   }
+
+   #This where you can find your SDA
+   return(list(Remote.Path = settings$outdir,
+               PID = PID))
+  
+}
+
+#' Remote_Sync_launcher
+#'
+#' @param settingPath Path to your local setting .
+#' @param remote.path Path generated by SDA_remote_launcher which shows the path to your remote SDA run.
+#' @param PID PID generated by SDA_remote_launcher which shows the active PID running your SDA job.
+#'
+#' @return
+#' @export
+#' @example 
+#' \dontrun{
+#'  library(PEcAn.all)
+#'  library(purrr)
+#'  
+#'  settingPath <-
+#'    "/fs/data3/hamzed/Projects/GeoTunnel/RemoteSDA/pecan.SDA.4sites.xml"
+#'  ObsPath <-
+#'    "/fs/data3/hamzed/Projects/GEF_MultiSite/Obs/LandTrendr_AGB_output50s.RData"
+#'  
+#'  
+#'  SDA_remote_info <-SDA_remote_launcher(settingPath, ObsPath)
+#'  
+#'  Remote_Sync_launcher(settingPath, SDA_remote_info$Remote.Path, SDA_remote_info$PID)
+#'  
+#'}
+Remote_Sync_launcher <- function(settingPath, remote.path, PID) {
+  
+  settings <- read.settings(settingPath)
+  
+  system(paste0("nohup Rscript ",
+                system.file("RemoteLauncher", "Remote_sync.R", package = "PEcAn.assim.sequential")," ",
+                settingPath, " ", 
+                remote.path, " ",
+                PID,
+                " > ",
+                settings$outdir,"/SDA_remote_report.out 2>&1 &"))
```

Comment:
Oh, I see....this is where the nohup comes in and is spawned by the SDA.  NOT the user initiating the SDA behind a nohup...which makes sense since if you were running localhost you would either nohup the workflow or run with model-launcher to run in the queue.